### PR TITLE
added new query for EC2 network ACL denied traffic

### LINKED ISSUE #1062: Add EC2 network ACL denied traffic query for cloudFormation
### Platform
cloudFormation

### Provider
cloudFormation

### Description
The deny rules defined for the selected AWS Network ACL are declared ineffective

### PR DESCRIPTION
**Pull request by soniantao**
Closes #1062
Closes #1063

### CODE REVIEW

File: assets/queries/cloudFormation/ec2_network_ACL_deny_rules/metadata.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "ec2_network_ACL_denied_traffic",
+  "queryName": "EC2 network ACL denied traffic",
+  "severity": "MEDIUM",
+  "category": null,
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/ec2_network_ACL_deny_rules/metadata.json:5]: `"category": "Network Security",`

File: assets/queries/cloudFormation/ec2_network_ACL_deny_rules/metadata.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  "descriptionText": "The deny rules defined for the selected AWS Network ACL are declared ineffective",
+  "descriptionUrl": "https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-network-acl-entry.html"
+}
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/ec2_network_ACL_deny_rules/metadata.json:8]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/ec2_network_ACL_deny_rules/query.rego [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  expectedAction := "deny"
+  contains(action,expectedAction) 
+  result := {
+                "documentId": 		input.document[i].id,
+                "searchKey": 	    sprintf("Resources.%s.Properties.CidrBlock", [name]),
+                "issueType":		"IncorrectValue",  
+                "keyExpectedValue": sprintf("'Resources.%s.Properties.CidrBlock should not be denied to traffic once RuleAction should be configured with allow.' ",[name]),
+                "keyActualValue": 	sprintf("'Resources.%s.Properties.CidrBlock will be denied to traffic since RuleAction is configured with deny.' ", [name])
+              }  
+ }
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/ec2_network_ACL_deny_rules/query.rego:19]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/ec2_network_ACL_deny_rules/test/negative.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  OutboundRule:
+    Type: AWS::EC2::NetworkAclEntry
+    Properties:
+       NetworkAclId:
+         Ref: MyNACL
+       RuleNumber: 100
+       Protocol: -1
+       Egress: true
+       RuleAction: allow
+       CidrBlock: 0.0.0.0/0
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/ec2_network_ACL_deny_rules/test/negative.yaml:30]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/ec2_network_ACL_deny_rules/test/positive.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  OutboundRule:
+    Type: AWS::EC2::NetworkAclEntry
+    Properties:
+       NetworkAclId:
+         Ref: MyNACL
+       RuleNumber: 100
+       Protocol: -1
+       Egress: true
+       RuleAction: deny
+       CidrBlock: 0.0.0.0/0
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/ec2_network_ACL_deny_rules/test/positive.yaml:30]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/ec2_network_ACL_deny_rules/test/positive_expected_result.json [hunk 6]
```diff
@@ -0,0 +1,7 @@
+[
+	{
+		"queryName": "EC2 network ACL denied traffic",
+		"severity": "MEDIUM",
+		"line": 17
+	}
+]
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/ec2_network_ACL_deny_rules/test/positive_expected_result.json:7]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/ec2_network_ACL_deny_rules/metadata.json [hunk 7]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "ec2_network_ACL_denied_traffic",
+  "queryName": "EC2 network ACL denied traffic",
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/ec2_network_ACL_deny_rules/metadata.json:3]: please change the **DIRECTORY name** and id to an all-lowercase string matching the query name: `ec2_network_acl_ineffective_denied_traffic`

change query name to: `EC2 Network ACL Ineffective Denied Traffic`

File: assets/queries/cloudFormation/ec2_network_ACL_deny_rules/query.rego [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+CxPolicy [result] {
+  resource := input.document[i].Resources[name]
+  resource.Type == "AWS::EC2::NetworkAclEntry"
+  action := resource.Properties.RuleAction
+  cidr := resource.Properties.CidrBlock
+  unvalidCidr := "0.0.0.0/0"
+  not contains(cidr,unvalidCidr)
+  expectedAction := "deny"
+  contains(action,expectedAction)
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/ec2_network_ACL_deny_rules/query.rego:11]: no need to declare a variable for this constant value: `contains(action,"deny")`

File: assets/queries/cloudFormation/ec2_network_ACL_deny_rules/query.rego [hunk 9]
```diff
@@ -0,0 +1,19 @@
+package Cx
+
+CxPolicy [result] {
+  resource := input.document[i].Resources[name]
+  resource.Type == "AWS::EC2::NetworkAclEntry"
+  action := resource.Properties.RuleAction
+  cidr := resource.Properties.CidrBlock
+  unvalidCidr := "0.0.0.0/0"
+  not contains(cidr,unvalidCidr)
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/ec2_network_ACL_deny_rules/query.rego:9]: simplify this block by replacing lines 8 and 9 with:
`not contains(cidr, "0.0.0.0/0")`

File: assets/queries/cloudFormation/ec2_network_ACL_deny_rules/metadata.json [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+  "severity": "MEDIUM",
+  "category": null,
+  "descriptionText": "The deny rules defined for the selected AWS Network ACL are declared ineffective",
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/ec2_network_ACL_deny_rules/metadata.json:6]: `"descriptionText": "Ineffective deny rules. A deny rule should be applied to all IP addresses.",`

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]
few things to fix

**Comment by rogeriopeixotocx** [comment]
LGTM ✅

**Review by joelsou5a — APPROVED** [review]

**Pull request merged**